File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        nodes = []
        temp = head
        while temp != None:
            nodes.append(temp)
            temp = temp.next
        if n == 0:
            nodes[len(nodes) - 2].next = None
            return head
        if n == len(nodes):
            head = head.next
            return head
        nodes[len(nodes) - n - 1].next = nodes[len(nodes) - n].next
        return head
        

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution 
{
    public boolean hasCycle(ListNode head) 
    {
        HashSet<Integer> seen = new HashSet<>();
        ListNode temp = head;
        if (temp == null){
            return false;
        }
        while (temp.next != null){
            if(!seen.add(temp.val)){
                return true;
            }
            temp = temp.next;
        }
        return false;
    }
}
